core/domain/user_services.py

```diff
@@ -1236,6 +1238,21 @@ def update_preferred_audio_language_code(
     _save_user_settings(user_settings)
 
 
+def update_preferred_text_language_code(
+        user_id, preferred_text_language_code):
+    """Updates preferred_text_language_code of user with given user_id.
+
+    Args:
+        user_id: str. The unique ID of the user.
+        preferred_text_language_code: str. New text translation language 
+            preference to set.
+    """
+    user_settings = get_user_settings(user_id, strict=True)
+    user_settings.preferred_text_language_code = (
+        preferred_text_language_code)
```

Comment:
Should we validate the supplied language code?

Reply:
I have added a validator in the `POST` argument of `TranslationPreferenceHandler`.